### PR TITLE
version 0.6.0

### PR DESCRIPTION
**Pull request by ThierryO**
Update palettes and themes

### CODE REVIEW

File: man/colour_blind_distance.Rd [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+\item{gray}{correct for monochromacy (unable to distinguish colours).
+Defaults to \code{FALSE}.}
+
+\item{fun}{Function to aggregate different colour distance matrices}
+
+\item{method}{passed to \code{colordistance::getColorDistanceMatrix()}}
+}
+\description{
+Calculate a distance matrix between colours for colour blind people
```
Comment (RCinbo) [inline man/colour_blind_distance.Rd:37]:
```suggestion
Calculate a distance matrix between colours for colour blind people.
```

File: R/colour_distance.R [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    as.data.frame() |>
+    cbind(Pct = 1)
+}
+
+#' Calculate a distance matrix between colours for colour blind people
+#' @param deutan correct for deuteranomaly (red - green colour blindness).
+#' Defaults to `TRUE`.
+#' @param protan correct for protanopia (red - green colour blindness).
+#' Defaults to `FALSE`.
+#' @param tritan correct for tritanopia (blue -yellow colour blindness).
```
Comment (RCinbo) [inline R/colour_distance.R:25]:
```suggestion
#' @param tritan correct for tritanopia (blue - yellow colour blindness).
```

File: man/colour_blind_distance.Rd (hunk 1 above)
Comment (RCinbo) [reply man/colour_blind_distance.Rd:37]: Couldn't find exactly where this line comes from in R/colour_distance.R

File: man/scale_fill_gradient.Rd [hunk 3]
```diff
@@ -2,7 +2,7 @@
 % Please edit documentation in R/scales.R
 \name{scale_fill_gradient}
 \alias{scale_fill_gradient}
-\title{redefine scale_fill_gradient}
+\title{redefine \code{ggplot2::scale_fill_gradient}}
```
Comment (RCinbo) [inline man/scale_fill_gradient.Rd:5]:
```suggestion
\title{redefine \code{ggplot2::scale_fill_gradient()}}
```

File: man/scale_viridis.Rd [hunk 4]
```diff
@@ -128,16 +129,25 @@ expand the scale by 5\% on each side for continuous variables, and by
 \item{alpha}{The alpha transparency, a number in [0,1], see argument alpha in
 \code{\link[grDevices]{hsv}}.}
 
-\item{begin}{The (corrected) hue in [0,1] at which the viridis colormap begins.}
+\item{begin, end}{The (corrected) hue in \verb{[0,1]} at which the color map
+begins and ends.}
 
-\item{end}{The (corrected) hue in [0,1] at which the viridis colormap ends.}
+\item{direction}{Sets the order of colors in the scale. If 1, the default,
+colors are ordered from darkest to lightest. If -1, the order of colors is
```
Comment (RCinbo) [inline man/scale_viridis.Rd:136]:
```suggestion
colors are ordered from darkest to lightest. If -1, the order of colors are
```

File: vignettes/child/_colour_test_figures.Rmd [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+\FloatBarrier
+
+```{r {{theme}}-histogram-oridinal, fig.cap = "Coloured along an ordinal variable."}
+ggplot(diamonds, aes(x = carat, y = depth, colour = cut)) +
+  geom_point()
+```
+
+\FloatBarrier
+
+```{r {{theme}}-categorical-2, fig.cap = "Coloured by a two level categorical variable"}
```
Comment (RCinbo) [inline vignettes/child/_colour_test_figures.Rmd:24]:
```suggestion
```{r {{theme}}-categorical-2, fig.cap = "Coloured by a two level categorical variable."}
```

File: vignettes/child/_colour_test_figures.Rmd [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+  x = runif(1e3), y = runif(1e3),
+  category = sample(LETTERS[1:2], 1e3, replace = TRUE)
+) |>
+  ggplot(aes(x = x, y = y, colour = category)) +
+  geom_jitter()
+```
+
+\FloatBarrier
+
+```{r {{theme}}-categorical-3, fig.cap = "Coloured by a three level categorical variable"}
```
Comment (RCinbo) [inline vignettes/child/_colour_test_figures.Rmd:35]:
```suggestion
```{r {{theme}}-categorical-3, fig.cap = "Coloured by a three level categorical variable."}
```

File: vignettes/child/_colour_test_figures.Rmd [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+  x = runif(1e3), y = runif(1e3),
+  category = sample(LETTERS[1:3], 1e3, replace = TRUE)
+) |>
+  ggplot(aes(x = x, y = y, colour = category)) +
+  geom_jitter()
+```
+
+\FloatBarrier
+
+```{r {{theme}}-categorical-4, fig.cap = "Coloured by a four level categorical variable"}
```
Comment (RCinbo) [inline vignettes/child/_colour_test_figures.Rmd:46]:
```suggestion
```{r {{theme}}-categorical-4, fig.cap = "Coloured by a four level categorical variable."}
```

File: vignettes/child/_colour_test_figures.Rmd [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+  x = runif(1e3), y = runif(1e3),
+  category = sample(LETTERS[1:4], 1e3, replace = TRUE)
+) |>
+  ggplot(aes(x = x, y = y, colour = category)) +
+  geom_jitter()
+```
+
+\FloatBarrier
+
+```{r {{theme}}-categorical-5, fig.cap = "Coloured by a five level categorical variable"}
```
Comment (RCinbo) [inline vignettes/child/_colour_test_figures.Rmd:57]:
```suggestion
```{r {{theme}}-categorical-5, fig.cap = "Coloured by a five level categorical variable."}
```

File: vignettes/child/_colour_test_figures.Rmd [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+  x = runif(1e3), y = runif(1e3),
+  category = sample(LETTERS[1:5], 1e3, replace = TRUE)
+) |>
+  ggplot(aes(x = x, y = y, colour = category)) +
+  geom_jitter()
+```
+
+\FloatBarrier
+
+```{r {{theme}}-categorical-6, fig.cap = "Coloured by a six level categorical variable"}
```
Comment (RCinbo) [inline vignettes/child/_colour_test_figures.Rmd:68]:
```suggestion
```{r {{theme}}-categorical-6, fig.cap = "Coloured by a six level categorical variable."}
```

File: vignettes/child/_colour_test_figures.Rmd [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+  x = runif(1e3), y = runif(1e3),
+  category = sample(LETTERS[1:6], 1e3, replace = TRUE)
+) |>
+  ggplot(aes(x = x, y = y, colour = category)) +
+  geom_jitter()
+```
+
+\FloatBarrier
+
+```{r {{theme}}-categorical-7, fig.cap = "Coloured by a seven level categorical variable"}
```
Comment (RCinbo) [inline vignettes/child/_colour_test_figures.Rmd:79]:
```suggestion
```{r {{theme}}-categorical-7, fig.cap = "Coloured by a seven level categorical variable."}
```

File: vignettes/child/_colour_test_figures.Rmd [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+  x = runif(1e3), y = runif(1e3),
+  category = sample(LETTERS[1:7], 1e3, replace = TRUE)
+) |>
+  ggplot(aes(x = x, y = y, colour = category)) +
+  geom_jitter()
+```
+
+\FloatBarrier
+
+```{r {{theme}}-categorical-8, fig.cap = "Coloured by an eight level categorical variable"}
```
Comment (RCinbo) [inline vignettes/child/_colour_test_figures.Rmd:90]:
```suggestion
```{r {{theme}}-categorical-8, fig.cap = "Coloured by an eight level categorical variable."}
```

File: vignettes/child/_themes_test_figures.Rmd [hunk 12]
```diff
@@ -1,76 +1,34 @@
...
-cancer$Proportion <- cancer$ncases / (cancer$ncases + cancer$ncontrols)
-```
-
-```{r {{theme}}-font, fig.cap = "Testing fonts"}
-d <- data.frame(
-  x = "Sphinx of black quartz, judge my vow",
-  y = "Sphinx of\nblack quartz,\njudge my vow"
-)
-ggplot(d, aes(x = x, y = x, label = y, colour = y)) +
+```{r {{theme}}-font, fig.cap = "Testing fonts", echo = TRUE}
```
Comment (RCinbo) [inline vignettes/child/_themes_test_figures.Rmd:1]:
```suggestion
```{r {{theme}}-font, fig.cap = "Testing fonts.", echo = TRUE}
```

File: vignettes/palettes.Rmd [hunk 13]
```diff
@@ -0,0 +1,291 @@
...
+
+```{r inbo-cluster, echo = FALSE, fig.cap = "Agglomerative hierarchical clustering of colours based on their distance."}
+library(cluster)
+inbo_palette() |>
+  colour_blind_distance() |>
+  agnes(method = "complete") |>
+  plot(ask = FALSE, which.plots = 2, main = "")
+```
+
+```{r inbo-palette-deutan, fig.cap = "Simulation of deutan model of colorblindness."}
```
Comment (RCinbo) [inline vignettes/palettes.Rmd:165]: colorblindness is written as two words in the title but as 1 word in the figure captions. I think both can be used but bette rnot to use them both in the same text. "color blindness" has more hits on google than "colorblindness"

File: vignettes/themes.Rmd [hunk 14]
```diff
@@ -10,159 +10,83 @@ vignette: >
...
-demo_palette(deutan(inbo_palette()))
-```
-
-```{r inbo-palette-protan, fig.cap = "Simulation of protan model of colorblindness."}
-demo_palette(protan(inbo_palette()))
-```
+The `ggplot2` package must be loaded prior to `INBOtheme`.
+Loading `INBOtheme` sets `inbo_theme()` as default theme and `inbo_steun_blauw` as default colour for the `geoms`.
+Note how we use `theme_set()` and `switch_colour()`.
+This applies the theme and colour to all subsequent plots within the need to change the `ggplot2` code.
```
Comment (RCinbo) [inline vignettes/themes.Rmd:23]:
```suggestion
This applies the theme and colour to all subsequent plots without the need to change the `ggplot2` code.
```

**Review by RCinbo — APPROVED** [review]
Nice work. I only have some minor remarks on spelling and textual consistency.

File: man/scale_viridis.Rd (hunk 4 above)
Comment (ThierryO) [reply man/scale_viridis.Rd:136]: This is inherited from the ggplot2 documentation.

File: man/scale_fill_gradient.Rd (hunk 3 above)
Comment (ThierryO) [reply man/scale_fill_gradient.Rd]: fixed in [8eda9d7](https://github.com/inbo/INBOtheme/pull/23/commits/8eda9d78996d7cf4427e783d09944464d18f8839)

**Comment by ElsLommelen** [comment]
While trying to install the package or build the vignette locally, I already notice that the INBO user is forced to install the newest version of `ggplot2` in order to be able to install the new version of `INBOtheme`.  Personally I update my package versions after each new R update at INBO, and for INBO packages I try to have backward compatibility at least to the point when the last R update was installed at INBO.  This is mainly to avoid problems for INBO users: updating specific packages (part of the installed packages) often gives compatibility problems and RStudio crashes, so forcing users to do so, may generate questions from users that experience crashes.  (They can of course also experience this with other packages, but a INBO package is likely to affect more INBO users.)  Certainly for starting R users, this kind of problems is often frustrating, as it is difficult to understand what they did wrong and how they can avoid this, so I can imagine this is one of the reasons why some users don't like using R and stop using R after a while.
So if not strictly needed, you may consider removing the argument `linewith` from functions `element_line()` and `element_rect()`.  And if not, you may want to provide a clear error message when the package installation fails, `unused argument (linewith = 0.5)` is not an error message that users associate with the need to update package ggplot2, I guess.  ;-)

**Comment by pietervsd** [comment]
I do not understand what you exactly mean in the introduction in the colour vignette.
In the themes vignette you show the different themes, but only with one colour, i think the default coloring along a variable is also a very important aspect of a theme.

File: vignettes/palettes.Rmd [hunk 15]
```diff
@@ -0,0 +1,291 @@
...
+```
+
+```{r vlaanderen-palette-demo}
+demo_palette(vlaanderen_palette())
+```
+
+## `traffic_palette()`
+
+Note that the `traffic_palette()` has a variable number of colours.
+It starts from a dark red colour and changes into a light green.
```
Comment (pietervsd) [inline vignettes/palettes.Rmd:58]:
```suggestion
It starts from a dark red colour and changes into a light green colour.
```

File: vignettes/palettes.Rmd [hunk 16]
```diff
@@ -0,0 +1,291 @@
...
+demo_palette(traffic_palette(3))
+```
+
+\FloatBarrier
+
+## `ordinal_palette()`
+
+`ordinal_palette()` has a variable number of colours.
+It starts with a dark colour and end with a light colour.
+Other colour are intermediate colours, resulting is a gradient from dark to light.
```
Comment (pietervsd) [inline vignettes/palettes.Rmd:83]:
```suggestion
Other colours are intermediate colours, resulting in a gradient from dark to light.
```

**Review by pietervsd — APPROVED** [review]

**Comment by ThierryO** [comment]
> While trying to install the package or build the vignette locally, I already notice that the INBO user is forced to install the newest version of `ggplot2` in order to be able to install the new version of `INBOtheme`.

`INBOtheme` requires `ggplot2 (>= 3.3.2)`. That requirement was set in ad16fb2 more than 2 years ago. The current PR doesn't force a more recent ggplot2 version. How were you forced to upgrade `ggplot2`?

**Comment by ElsLommelen** [comment]
> > While trying to install the package or build the vignette locally, I already notice that the INBO user is forced to install the newest version of `ggplot2` in order to be able to install the new version of `INBOtheme`.
> 
> `INBOtheme` requires `ggplot2 (>= 3.3.2)`. That requirement was set in [ad16fb2](https://github.com/inbo/INBOtheme/commit/ad16fb26e24fc60e05b96bbeda8ecd2bbbb6f1d3) more than 2 years ago. The current PR doesn't force a more recent ggplot2 version. How were you forced to upgrade `ggplot2`?

When installing this branch using `remotes::install_github("inbo/INBOtheme@palette")`, I got the following error:

```
Error: package or namespace load failed for 'INBOtheme':
 .onLoad failed in loadNamespace() for 'INBOtheme', details:
  call: element_line(colour = "black", linewidth = 0.5, linetype = 1, 
  error: unused argument (linewidth = 0.5)
Error: loading failed
Execution halted
ERROR: loading failed
* removing 'C:/R/library/INBOtheme'
* restoring previous 'C:/R/library/INBOtheme'
```

Earlier in the installation, I also got the following notes:
```
** byte-compile and prepare package for lazy loading
Note: possible error in 'element_line(colour = "black", ': unused argument (linewidth = 0.5) at theme_elsevier.R:23 
Note: possible error in 'element_rect(fill = "white", ': unused argument (linewidth = 0.5) at theme_elsevier.R:23 
Note: possible error in 'element_line(colour = "grey90", ': unused argument (linewidth = 0.2) at theme_elsevier.R:23 
Note: possible error in 'element_line(colour = "grey98", ': unused argument (linewidth = 0.5) at theme_elsevier.R:23 
Note: possible error in 'element_line(colour = "black", ': unused argument (linewidth = 0.5) at theme_inbo2015.R:60 
Note: possible error in 'element_rect(fill = rect_bg, ': unused argument (linewidth = 0.5) at theme_inbo2015.R:60 
Note: possible error in 'element_line(colour = panel_grid, ': unused argument (linewidth = 0.25) at theme_inbo2015.R:60 
Note: possible error in 'element_line(colour = "black", ': unused argument (linewidth = 0.5) at theme_vlaanderen2015.R:56 
Note: possible error in 'element_rect(fill = rect_bg, ': unused argument (linewidth = 0.5) at theme_vlaanderen2015.R:56 
Note: possible error in 'element_line(colour = panel_grid, ': unused argument (linewidth = 0.25) at theme_vlaanderen2015.R:56 
** help
```

My local documentation in `?ggplot2::element_line` doesn't mention an argument `linewidth`, whereas [this website](https://ggplot2.tidyverse.org/reference/element.html) does.  I am using ggplot version 3.3.6, which is installed after R 4.2.1 was installed at INBO, and apparently the argument `linewidth` is added in ggplot version 3.4.0.  So in fact INBOtheme should require `ggplot2 (>= 3.4.0)` when using linewidth, or even `ggplot2 (>= 3.4.1)` when using it in a theme, [I guess](https://ggplot2.tidyverse.org/news/index.html).  Now the user gets a very strange error during installation, which is due to the use of very recent features and not requiring to have this recent version.  And indirectly, due to the installation error, a user is forced to upgrade ggplot2 anyway...

**Comment by ThierryO** [comment]
> I do not understand what you exactly mean in the introduction in the colour vignette. In the themes vignette you show the different themes, but only with one colour, i think the default coloring along a variable is also a very important aspect of a theme.

My idea was to make the individual vignettes smaller. The 'themes' vignette focusses on the styling and the default colour of geoms visualised with different kinds of geoms.  The 'colours' vignette focusses on colours a long a continuous, ordinal or discrete variable.

Maybe I should add an introduction to the 'colours' vignette like the one in the 'themes' vignette.

File: vignettes/themes.Rmd [hunk 17]
```diff
@@ -10,159 +10,83 @@ vignette: >
...
 
-# Available themes
+# Default: `theme_inbo()`
 
-## Default: `theme_inbo`
+## `theme_inbo()` with default background
 
 ```{r set_theme_inbo, echo = TRUE}
-theme_set(theme_inbo())
+theme_set(theme_inbo(7))
```
Comment (ElsLommelen) [inline vignettes/themes.Rmd:89]:
```suggestion
theme_set(theme_inbo(base_size = 7))
```
It is unclear for a reader why this `7` is there.  As suggested, just adding the argument already solves my concern.
On the other hand, as you seem to be using the 7 everywhere, it seems this is a perfect size for a default, or am I mistaken here?

**Review by ElsLommelen — COMMENTED** [review]
I just had a quick view at the code itself (without building the vignette), and except for the remark of @pietervsd on the introduction of the colour vignette being unclear, I only have a minor remark (but the same suggestion can be given on more than one code example).

File: vignettes/themes.Rmd (hunk 17 above)
Comment (ThierryO) [reply vignettes/themes.Rmd:89]: A good `base_size` depends on the output format. See https://github.com/inbo/INBOmd/blob/33fdf6c06ec1f357ba048c5f24831726be2ea3b1/inst/rmarkdown/templates/report/skeleton/skeleton.Rmd#L118 for a few examples. The current default works best for interactive use.

Comment (ElsLommelen) [reply vignettes/themes.Rmd:89]: Ok, in this case just consider adding `base_size =` to make clear for the user where the `7` stands for.  (I checked it in the function documentation, but as a reader I would rather like it to be clear in the vignette itself.)

**Pull request merged**